packages/clay-data-provider/src/LRUCache.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const DEFAULT_MAX_SIZE = 20;
+
+export class LRUCache<K, V> {
+	maxSize: number;
+	storage: Map<K, V>;
+
+	constructor(maxSize: number = DEFAULT_MAX_SIZE) {
+		if (maxSize <= 0) {
+			throw new TypeError('LRUCache: max must be a non-negative number');
+		}
+
+		this.maxSize = maxSize;
+		this.storage = new Map();
+	}
+
+	reset() {
+		this.storage = new Map();
+		return this.storage;
+	}
+
+	has(key: K) {
+		return this.storage.has(key);
+	}
+
+	get(key: K) {
+		const value = this.storage.get(key);
+
+		if (value) {
+			this.storage.delete(key);
+			this.storage.set(key, value);
+		}
+
+		return value;
+	}
+
+	set(key: K, value: V) {
+		if (this.storage.size >= this.maxSize) {
+			const iterator = this.storage.keys();
+			let count = this.maxSize - (this.storage.size - 1);
+			while (count) {
+				const {value} = iterator.next();
+				this.storage.delete(value);
+				count--;
+			}
+		}
+
+		this.storage.set(key, value);
+		return true;
```

Comment:
Do you really want to return `true` here? It seems pointless to have a function that always returns `true` no matter what. Given that the cache is basically a `Map`, we could return `this` like `Map.prototype.set` does (that's what I did in [mine](https://github.com/wincent/masochist/blob/8ee6fc3704b204adae6dc3ad9baca5108a741637/src/server/LRUCache.js#L52) — this means you can chain multiple calls; eg. `new LRUCache().set('a', 1).set('b', 2)` — or just `undefined` (that's what [the other one](https://github.com/facebook/relay/commit/3f5d799fa8beb5f810f2bd37813674fc22598b74#diff-37a526caff576f11bac88a48c97391ebR49) I linked you to does).

Looking at how you are using this return value, I see it goes through many layers of indirection inside `useCache` and ultimately winds up discarded inside `useResource` when it does `cache.set(result)` and ignores the return value.

Reply:
Hmm, good question, this would probably be a trade-off in returning `true` or `undefined` in the end would be the same as returning `true` even if you don't use it but the purpose is just another meaning of showing that everything will always work out and Assuming that the method actually did is `true` then `true`, `1`... it would make little sense to return `undefined`, `NuN`, `false` because they are treated as negative or nothing happened etc...

I think in the end I prefer to keep the chainable, it seems to be more useful but maybe not use here, so I prefer to go with the `undefined` return here to be in the middle ground 🙂.